cpp/dolfinx/mesh/MeshTags.h

```diff
@@ -75,6 +75,26 @@ class MeshTags
   /// Move assignment
   MeshTags& operator=(MeshTags&& tags) = default;
 
+  /// Find all MeshTag indices of a given value, local-to-process
+  /// @param[in] i The value
+  /// @return The indices
+  std::vector<std::int32_t> find_indices(T i) const
+  {
+    int n = std::count(_values.begin(), _values.end(), i);
```

Comment:
I would do rather
```cpp
std::vector<std::int32_t> indices;
indices.reserve(_indices.size());
for (std::int32_t i = 0; i < _values.size(); ++i)
{
  if (_values[i] == value)
    indices.push_back(_indices[i]);
}
return indices;
```

Reply:
That could use a lot more memory than required. Perhaps:
```
int n = std::count(_values.begin(), _values.end(), i);
indices.reserve(n);
for (std::int32_t i = 0; i < _values.size(); ++i)
  if (_values[i] == value)
    indices.push_back(_indices[i]);

return indices;
```

Reply:
Maybe we should go without fancy `reserve`. I think `std::count()` has surely complexity O(n) in container size which is also not good. Just a pure push_back has sufficiently clever preallocation strategy.

Reply:
Clever -> potentially large memory overhead, needing `shrink` afterwards. The search is already O(n), so preallocating doesn't affect complexity and I expect will be faster.